test/cppsuite/test_harness/core/configuration.h

```diff
@@ -173,6 +181,100 @@ class configuration {
         return func(value);
     }
 
+    /*
+     * Merge together two configuration strings, the user one and the default one.
+     */
+    std::string
+    merge_default_config(const std::string &default_config, const std::string &user_config)
+    {
+        std::string merged_config;
+        auto split_default_config = split(default_config);
+        auto split_user_config = split(user_config);
+        auto user_it = split_user_config.begin();
+        for (auto it = split_default_config.begin(); it != split_default_config.end(); ++it) {
+            if (user_it->first != it->first)
+                /* The default does not exist in the user configuration, add it. */
+                merged_config += it->first + "=" + it->second;
+            else {
+                /* If we have a sub config merge it in. */
+                if (user_it->second[0] == '(')
+                    merged_config +=
+                      it->first + "=(" + merge_default_config(it->second, user_it->second) + ')';
+                else
+                    /* Add the user configuration as it exists. */
+                    merged_config += user_it->first + "=" + user_it->second;
+                user_it++;
+            }
+            /* Add a comma after every item we add except the last one. */
+            if (split_default_config.end() - it != 1)
+                merged_config += ",";
+        }
+        return (merged_config);
+    }
+
+    /*
+     * Split a config string into keys and values, taking care to not split incorrectly when we have
+     * a sub config.
+     */
+    static std::vector<std::pair<std::string, std::string>>
+    split(const std::string &config)
+    {
+        std::string cut_config = config;
+        std::vector<std::pair<std::string, std::string>> split_config;
+        std::string key = "", value = "";
+        bool in_subconfig = false;
+        bool expect_value = false;
+        std::stack<char> subconfig_parens;
+
+        /* Remove prefix and trailing "()". */
+        if (config[0] == '(')
+            cut_config = config.substr(1, config.size() - 2);
```

Comment:
What if config is of size 1 ?

Reply:
Good catch, I'll add some edge case checking